### PR TITLE
[ED] Implement usage of disclosed contracts in command interpretation [DPP-1096]

### PR DESCRIPTION
**Pull request by tudor-da**
Implement usage of disclosed contracts in command interpretation

* NOTE: No actual disclosed contracts can be passed at this point to command interpretation since command validation disallows it

changelog_begin
changelog_end

### CODE REVIEW

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala [hunk 1]
```diff
@@ -86,6 +86,7 @@ final class CommandsValidator(ledgerId: LedgerId) {
         ledgerEffectiveTime = ledgerEffectiveTimestamp,
         commandsReference = workflowId.fold("")(_.unwrap),
       ),
+      disclosedContracts = ImmArray.empty,
```
Comment (tudor-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala:89]: Disclosed contracts validation will be addressed in a follow-up PR.

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  it should "forward contract store lookup result on archived contracts" in new TestScope {
+    private val actualResult: MaximumLedgerTime =
+      resolveMaximumLedgerTime(ImmArray.empty, archived.contracts + cId_1).futureValue
+
+    actualResult match {
+      case `archived` => ()
+      case other => fail(s"Unexpected $other")
+    }
```
Comment (skisel-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala:82]:
```suggestion
    resolveMaximumLedgerTime(
      ImmArray.empty,
      archived.contracts + cId_1,
    ).futureValue shouldBe archived
```

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  it should "resolve maximum ledger time when no disclosed contracts are provided" in new TestScope {
+    private val actualResult: MaximumLedgerTime =
+      resolveMaximumLedgerTime(ImmArray.empty, Set(cId_1, cId_2)).futureValue
+
+    actualResult match {
+      case MaximumLedgerTime.Max(actualLedgerTime) => actualLedgerTime shouldBe t2
+      case other => fail(s"Unexpected $other")
+    }
```
Comment (skisel-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala:72]:
```suggestion
    resolveMaximumLedgerTime(
      ImmArray.empty,
      Set(cId_1, cId_2),
    ).futureValue shouldBe MaximumLedgerTime.Max(t2)
```

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala (hunk 2 above)
Comment (skisel-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala:82]: Let me know if I missed something :)

**Review by skisel-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/ResolveMaximumLedgerTime.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  * @param contractStore The contract store.
+  */
+class ResolveMaximumLedgerTime(contractStore: ContractStore) {
+  def apply(
+      usedDisclosedContracts: ImmArray[DisclosedContract],
+      usedContractIds: Set[ContractId],
+  )(implicit lc: LoggingContext): Future[MaximumLedgerTime] = {
+    val usedDisclosedContractIds = usedDisclosedContracts.iterator.map(_.contractId).toSet
+
+    val contractIdsToBeLookedUp = usedContractIds -- usedDisclosedContractIds
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/ResolveMaximumLedgerTime.scala:29]: Can you remind me what have we decided to do with the liars, who send disclosedContracts in a command lying about the age (or other aspects), but correct information is present on the IndexDb

Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/ResolveMaximumLedgerTime.scala:29]: The Daml Engine filters them out so here we are already only handling DCs that are guaranteed to be used.

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  it should "resolve maximum ledger time when all contracts are provided as explicitly disclosed" in new TestScope {
+    private val disclosedContracts = ImmArray(
+      buildDisclosedContract(cId_1, t1),
+      buildDisclosedContract(cId_2, t2),
+    )
+
+    private val actualResult: MaximumLedgerTime =
+      resolveMaximumLedgerTime(disclosedContracts, Set(cId_1, cId_2)).futureValue
+
+    actualResult match {
```
Comment (skisel-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/execution/ResolveMaximumLedgerTimeSpec.scala:59]: actualResult shouldBe MaximumLedgerTime.Max(t2)

**Pull request merged**